src/data-sources/DataSourceList.tsx

```diff
@@ -152,8 +149,18 @@ const DataSourceList = () => {
 		table: {},
 	};
 
-	const isItemEligibleForActions = ( item: DataSourceConfig ) => {
-		return item.config_source === ConfigSource.STORAGE;
+	const isItemEligibleForActions = ( item: DataSourceConfig, action: string ) => {
```

Comment:
What I have done is, using the enabled status to only allow delete and copy actions. Everything else is disabled.